### PR TITLE
Raise EtcdValueErrors on CAS errors

### PR DESCRIPTION
**Pull request by alexhersh**
Narrow Exceptions to etcd specific handlers.

### CODE REVIEW

File: calico_containers/pycalico/ipam.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def _compare_and_swap_block(self, block):
             _log.debug("CAS Update block %s", block)
             try:
                 self.etcd_client.update(block.update_result())
-            except ValueError:
+            except EtcdValueError:
```
Comment (caseydavenport) [inline calico_containers/pycalico/ipam.py]: To be as specific as possible, shouldn't these all be catching `EtcdCompareFailed`?

Comment (alexhersh) [inline calico_containers/pycalico/ipam.py]: I was worried that ValueError would catch more than the Compare Error that I expected, but looking again at the Exception hierarchy, I think you are right, we can afford to be more specific.

File: calico_containers/pycalico/ipam.py [hunk 2]
```diff
@@ -322,7 +322,7 @@ def _compare_and_swap_handle(self, handle):
                 _log.debug("Handle %s is not empty.", handle.handle_id)
                 try:
                     self.etcd_client.update(handle.update_result())
-                except ValueError:
+                except EtcdCompareFailed:
                     raise CASError(handle.handle_id)
         else:
             _log.debug("CAS Write new handle %s", handle.handle_id)
```
Comment (caseydavenport) [inline calico_containers/pycalico/ipam.py:328]: Will write(...) throw an EtcdCompareFailed?  Do we need to catch it below?

Comment (alexhersh) [inline calico_containers/pycalico/ipam.py:328]: No, I think they only trigger when the block has a db_result to compare against (update not write).

**Comment by caseydavenport** [comment]
LGTM

**Pull request merged**